### PR TITLE
Bad row for illegal querystring

### PR DESCRIPTION
**Pull request by dilyand**

### CODE REVIEW

**Comment by dilyand** [comment]
Pre-changes:

```
| Ⰱ :> curl -X GET -i http://localhost:12345/i?"a=%b"
HTTP/1.1 500 Internal Server Error
Server: akka-http/10.1.12
Date: Wed, 09 Dec 2020 14:26:17 GMT
Content-Type: text/plain; charset=UTF-8
Content-Length: 74

Illegal query: Unexpected end of input, expected HEXDIG (line 1, column 5)
```

> [ERROR] [12/09/2020 14:26:17.104] [scala-stream-collector-akka.actor.default-dispatcher-3] [akka.actor.ActorSystemImpl(scala-stream-collector)] Error during processing of request: 'Illegal query: Unexpected end of input, expected HEXDIG (line 1, column 5): a=%b
    ^'. Completing with 500 Internal Server Error response. To change default exception handling behavior, provide a custom ExceptionHandler.
akka.http.scaladsl.model.IllegalUriException: Illegal query: Unexpected end of input, expected HEXDIG (line 1, column 5): a=%b
    ^
	at akka.http.scaladsl.model.IllegalUriException$.apply(ErrorInfo.scala:74)
        (...)

Post-changes:

```
| Ⰱ :> curl -X GET -i http://localhost:12345/i?"a=%b"
HTTP/1.1 200 OK
Server: akka-http/10.1.12
Date: Wed, 09 Dec 2020 14:25:04 GMT
Content-Length: 0
```

> [DEBUG] [12/09/2020 14:25:04.581] [scala-stream-collector-akka.actor.default-dispatcher-9] [akka://scala-stream-collector/system/IO-TCP/selectors/$a/0] New connection accepted
eyJwcm9jZXNzb3IiOnsiYXJ0aWZhY3QiOiJzbm93cGxvdy1zdHJlYW0tY29sbGVjdG9yIiwidmVyc2lvbiI6IjIuMS4wLXFzIn0sImZhaWx1cmUiOnsidGltZXN0YW1wIjoiMjAyMC0xMi0wOVQxNDoyNTowNC42MTI4MzJaIiwiZXJyb3JzIjpbIklsbGVnYWwgcXVlcnk6IFVuZXhwZWN0ZWQgZW5kIG9mIGlucHV0LCBleHBlY3RlZCBIRVhESUcgKGxpbmUgMSwgY29sdW1uIDUpOiBhPSViXG4gICAgXiJdfSwicGF5bG9hZCI6ImE9JWIifQ==

File: build.sbt [hunk 1]
```diff
@@ -69,7 +69,7 @@ lazy val root = project.in(file(".")).settings(buildSettings).aggregate(core, ki
 
 lazy val core = project
   .settings(moduleName := "snowplow-stream-collector-core")
-  .settings(buildSettings)
+  .settings(buildSettings ++ BuildSettings.sbtAssemblySettings)
```
Comment (chuwy) [inline build.sbt:72]: We've added assembly, but where's the logic of actual publishing? Just FYI we shoudn't publish jars on S3 anymore - GH assets should be preferred https://github.com/snowplow-incubator/snowplow-bigquery-loader/issues/135#issuecomment-741006011

**Review by chuwy — APPROVED** [review]
I would personally extract logic of `Right` into a separate function as its quite big, but the current version also looks good.

File: core/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala [hunk 2]
```diff
@@ -312,6 +357,20 @@ class CollectorServiceSpec extends Specification {
...
+      "write out the generated bad row" in {
+        val br = BadRow.GenericError(
+          Processor(generated.BuildInfo.name, generated.BuildInfo.version),
+          Failure.GenericFailure(Instant.now(), NonEmptyList.one("IllegalQueryString")),
+          Payload.RawPayload("")
+        )
+        val l = service.sinkBad(br, "key")
+
+        l must have size 1
+        l.head.zip(br.asJson.noSpaces).forall { case (a, b) => a mustEqual b }
```
Comment (benjben) [inline core/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala:370]:
```suggestion
        l ====  List(br.asJson.noSpaces.getBytes())
```

**Review by benjben — APPROVED** [review]
:+1:

**Pull request merged**